sotodlib/tod_ops/jumpfind.py

```diff
@@ -0,0 +1,406 @@
+                     something non-physical.
+
+        max_depth: The maximum recursion depth, set negetive to not use
+
+        depth: The current recursion depth
+
+    Returns:
+
+        jumps: The indices of jumps in x
+               There is some uncertainty on order of a few samples
+               Jumps within 20 samples of each other may not be distinguished
+    """
+    _x = sig.detrend(x)
+    if _x.std() > 10:
+        _x = _x / (10 ** (int(np.log10(_x.std()))))
+    if np.isclose(_x.std(), 0.0):
+        return np.array([])
+
+    jumps = jumpfind_gaussian(
+        _x,
+        2 * _x.std(),
+        0,
+        10,
+        1.0 / sensitivity,
+        0,
+        20,
+        0,
+        height=1,
+        prominence=1,
+    )
+
+    # If no jumps found return
+    if len(jumps) == 0:
+        return jumps
+
+    # If at max_depth then return
+    if depth == max_depth:
+        return jumps
+
+    # Recursively check for jumps between jumps
+    _jumps = np.insert(jumps, 0, 0)
+    _jumps = np.append(_jumps, len(x))
+    added = 0
+    for i in range(len(_jumps) - 1):
+        sub_jumps = jumpfind_recursive_gaussian(
+            x[(_jumps[i]) : (_jumps[i + 1])], sensitivity, max_depth, depth + 1
+        )
+        jumps = np.insert(jumps, i + added, sub_jumps + _jumps[i])
+        added += len(sub_jumps)
+    return jumps
+
+
+def jumpfind_sliding_window(
+    x, window_size=10000, overlap=1000, jumpfinder=jumpfind_tv, **kwargs
+):
+    """
+    Run jumpfinder through a sliding window.
+    This can help get jumps towards the edges of the data that may be missed.
+    Nominally those jumps can be found if the jumpfinder reaches sufficient depth,
+    but sometimes it takes tweaking of the parameters to catch them.
+
+    Arguments:
+
+        x: Data to jumpfind on, expects 1D
+
+        window_size: Size of window to use
+
+        overlap: Overlap between adjacent windows
+
+        jumpfinder: Jumpfinding function to use
+
+        **kwargs: Keyword args to pass to jumpfinder
+
+    Returns:
+
+        jumps: The indices of jumps in x
+               There is some uncertainty on order of 1 sample
+               Jumps within min_chunk of each other may not be distinguished
+    """
+    jumps = np.array([], dtype=int)
+    for i in range(len(x) // (window_size - overlap)):
+        start = i * (window_size - overlap)
+        end = np.min((start + window_size, len(x)))
+        _jumps = jumpfinder(x[start:end], **kwargs) + start
+        jumps = np.hstack((jumps, _jumps))
+    return np.unique(jumps).astype(int)
+
+
+def jumpfind(tod, signal=None, buff_size=10, jumpfinder=jumpfind_tv, **kwargs):
+    """
+    Find jumps in tod.signal_name.
+    Expects tod.signal_name to be 1D of 2D
+
+    Arguments:
+
+        tod: axis manager
+
+        signal: Signal to jumpfind on. If None than tod.signal is used.
+
+        buff_size: How many samples to flag around each jump in RangesMatrix 
```

Comment:
The choice of 10 for the default seems very arbitrary.  I think this should be off by default, so the user just gets single-sample jump locations, or else the default should be to have it computed based on other function parameters (such as the aforementioned "time_scale" for filtering/windowing).

Reply:
10 was just because typically the slop in the position of the jump is less than 10 samples. But this uncertainty is mostly only an issue in super jumpy regions (where jumps right next to each other aren't always distinguished). It's also less of an issue with the TV filtering, the gaussian adds some width to the jumps when it smooths so you can get some additional error.